src/azureExtensions.ts

```diff
@@ -3,19 +3,16 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { GenericResource } from "@azure/arm-resources";
-import { isAppServiceApp, isFunctionApp } from "./utils/azureUtils";
+import { AzExtResourceType } from "@microsoft/vscode-azext-utils/azExtResourceType";
+import { AppResource } from "@microsoft/vscode-azext-utils/hostapi";
 import { localize } from "./utils/localize";
 
 export const azureExtensions: IAzExtMetadata[] = [
     {
         name: 'vscode-azurefunctions',
         label: 'Functions',
         resourceTypes: [
-            {
```

Comment:
These changes will affect the extension activation manager, since the Azure resource type is used as the qualifier for activation. I agree that the `resourceType` field should be changed here as you have it, but I think we need to also include a separate `azureResourceType` (or some name along those lines), that the extension activation manager can use, _in addition_ to using the new `AzExtResourceType`--so the activation manager would activate _both_ (for now). Eventually we can remove the Azure resource type from here and the activation manager when all of the extensions have shipped.

Unless we're absolutely sure we're going to synchronously ship all of them again...

Reply:
We should discuss this in standup.

There are a few more breaking changes I'd like to make based on `AzExtResourceType`. So I was hoping we could sim-ship the extensions.

I'll look into making the changes backwards compatible too.

Reply:
> These changes will affect the extension activation manager, since the Azure resource type is used as the qualifier for activation.

I agree we should change the extension activation manager (EAM) to use `AzExtResourceType`. However, these changes don't actually break the existing EAM implementation because each extension [declares the Azure type in the the manifest](https://github.com/microsoft/vscode-azureresourcegroups/blob/2a19e3397229f75e95eb39fcbfa5d032c7b32446/src/utils/ExtensionActivationManager.ts#L56-L57).

Reply:
*smack* Oh duh! Yeah. we should try to set up EAM to try waking up both the old `microsoft.blah/blah` types and the new `AzExtResourceType`, over time we can convert the extensions to have the new items.